server/src/main/java/io/deephaven/server/session/SessionState.java

```diff
@@ -648,7 +651,9 @@ private synchronized void setDependencies(final List<ExportObject<?>> parents) {
 
             this.parents = parents;
             dependentCount = parents.size();
-            parents.stream().filter(Objects::nonNull).forEach(this::tryManage);
+            alreadyDeadParent = parents.stream()
```

Comment:
Should we unmanage all parents we successfully managed, if `alreadyDeadParent` is not null? Are we doing ourselves any favors by preserving extra parents to support this node that cannot succeed?